test/qir/runtime/CMakeLists.txt

```diff
@@ -13,7 +13,10 @@ macro(ADD_QIR_CIRCUIT target_name circuit_path)
     get_filename_component(circuit_name ${circuit_path} NAME_WE)
     add_custom_command(
       OUTPUT ${circuit_name}.o
-      COMMAND clang -c ${circuit_path} -o ${circuit_name}.o
+      COMMAND
+        clang
+        "$<$<BOOL:${CMAKE_OSX_DEPLOYMENT_TARGET}>:-mmacos-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}>"
+        -c ${circuit_path} -o ${circuit_name}.o
```

Comment:
Just because I now see this here: why are we explicitly calling `clang` here for this? This seems fragile to me. Why is that not a regular CMake target?
(sorry for the off-topic comment here)